src/libraries/System.Private.CoreLib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -262,4 +262,47 @@ public void GetObjectData(SerializationInfo info, StreamingContext context)
         public override int GetHashCode() =>
             GetType().GetHashCode();
     }
+
+    // This class exists to be EqualityComparer<string>.Default. It can't just use the GenericEqualityComparer<string>,
+    // as it needs to also implement IAlternateEqualityComparer<ReadOnlySpan<char>, string>, and it can't be
+    // StringComparer.Ordinal, as that doesn't derive from the abstract EqualityComparer<T>.
```

Comment:
Would changing this be considered too breaking here? I don't think many people would rely on it being `IEqualityComparer` but not `EqualityComparer`.

Reply:
Would changing what be too breaking?

Reply:
I mean making `StringComparer.Ordinal` derrive from `EqualityComparer`.

Reply:
The type backing StringComparer.Ordinal must derive from StringComparer; changing that would be a massive breaking change we'd never do. It's possible the abstract base class StringComparer itself could be changed to derive from `EqualityComparer<string>`; I've not explored what ramifications there might be from that.

> I don't think many people would rely on it being IEqualityComparer but not EqualityComparer.

Sorry, not following. Tons of code relies on the interface rather than the abstract class.

Reply:
>> It's possible the abstract base class StringComparer itself could be changed to derive from EqualityComparer<string>; I've not explored what ramifications there might be from that.

> Yeah that's what I've meant here

It could be explored separately if there's good reason to. Not worth it impacting public surface area for this.